resources/ts/degree-program-overview/degree-program-overview.ts

```diff
@@ -0,0 +1,92 @@
+import { _x } from '@wordpress/i18n';
+
+import DegreeProgram, { DegreeProgramApiData } from './degree-program';
+
+const DEGREE_PROGRAMS_SECTION_SELECTOR = '.c-degree-programs-search';
+const DEGREE_PROGRAMS_OVERVIEW_SELECTOR = '.c-degree-programs-collection';
+const SINGLE_PROGRAM_PREVIEW_SELECTOR = '.c-degree-program-preview';
+const NO_SEARCH_RESULT_SELECTOR = '.c-no-search-results';
+
+const degreeProgramsSection = document.querySelector< HTMLElement >(
+	DEGREE_PROGRAMS_SECTION_SELECTOR
+);
+const degreeProgramsOverview =
+	degreeProgramsSection?.querySelector< HTMLElement >(
+		DEGREE_PROGRAMS_OVERVIEW_SELECTOR
+	);
+
+export const currentLanguage =
+	degreeProgramsSection?.getAttribute( 'lang' )?.substring( 0, 2 ) || 'de';
+
+const renderPrograms = ( programs: DegreeProgram[] ) => {
+	const output = programs.map( ( program ) => program.render() ).join( '' );
+	degreeProgramsOverview?.insertAdjacentHTML( 'beforeend', output );
+};
+
+const renderNoResults = () => {
+	const output = `
+		<p class="c-no-search-results">
+			${ _x(
+				'No degree programs found',
+				'frontoffice: Search results',
+				'fau-degree-program-output'
+			) }
+		</p>`;
+	degreeProgramsSection?.insertAdjacentHTML( 'beforeend', output );
+};
+
+const showNoResults = () => {
+	const noResults = degreeProgramsSection?.querySelector(
+		NO_SEARCH_RESULT_SELECTOR
+	) as HTMLElement;
+
+	if ( ! noResults ) {
+		renderNoResults();
+
+		return;
+	}
+
+	noResults.classList.remove( 'hidden' );
+};
+
+const hideNoResults = () => {
+	const noResults = degreeProgramsSection?.querySelector(
+		NO_SEARCH_RESULT_SELECTOR
+	) as HTMLElement;
+
+	if ( ! noResults ) {
+		return;
+	}
+
+	noResults.classList.add( 'hidden' );
+};
+
+export const updateDegreeProgramOverviewDataset = (
+	dataset: Record< string, any >
+) => {
+	if ( ! ( degreeProgramsOverview instanceof HTMLElement ) ) {
+		return;
+	}
+
+	Object.entries( dataset ).forEach( ( [ property, value ] ) => {
+		degreeProgramsOverview.dataset[ property ] = value;
+	} );
+};
+
+export default ( data: DegreeProgramApiData[] ) => {
```

Comment:
IMO this is the place to add all the a11y-related stuff. I would definitely add the `aria-busy` attribute to indicate when the results loading starts and ends. You can also mark the results wrapper as a live region by adding a `role="region"` and `aria-live="polite"` so that dynamic content changes are announced to screen readers.